[Clang] unrecognized html tag causing undesirable comment lexing

mdenson opened the pull request:
Simple fix for this particular html tag. A more complete solution should be implemented. 

1. Add all html tags to table so they are recognized. Some input on what is desirable/safe would be appreciated
2. Change the lex strategy to deal with this in a different manner

Fixes #32680

brock-vircient committed:
[clang] unrecognized html tag causing undesirable comment lexing. fixes #32680

brock-vircient committed:
[clang] unrecognized html tag causing undesirable comment lexing. fixes #32680

cor3ntin commented on clang/test/AST/ast-dump-comment.cpp:
```diff
@@ -131,3 +131,12 @@ void Test_TemplatedFunctionVariadic(int arg, ...);
+// CHECK-NEXT:    ParagraphComment
+// CHECK:           HTMLStartTagComment{{.*}} Name="summary"
+// CHECK-NEXT:        TextComment{{.*}} Text="Short summary"
+// CHECK-NEXT:        HTMLEndTagComment{{.*}} Name="summary"
```
Missing line break

cor3ntin reviewed (commented):
This change needs a release note.
Please add an entry to `clang/docs/ReleaseNotes.rst` in the section the most adapted to the change, and referencing any Github issue this change fixes. Thanks!

mdenson commented:
Added newline and updated release note.

I also added, what I thought might be some other reasonable tags:
- details (used with summary)
- mark (for highlighting)
- figure and figcaption

I can back those out if needed.

I didn't write tests for each individually, but I can if needed.

mdenson committed:
Merge branch 'main' into issue-32680

mdenson commented:
let me know if you want the extra changes backed out. otherwise, you will need to merge it since i don't have access.

ojhunt reviewed (commented):
The html tag updates looks good to me - tags are correctly described.

There should be tests for each tag however. 

As @cor3ntin has reviewed previously, I'd like his sign off - but as a question to him: do we expect tests for invalid/unclosed tags? my default is to want such tests but that's entirely due to my former work :D

brock-vircient committed:
add tests for new tags

ojhunt commented:
> sorry guys, doing too many things at once. just that last commit is mine.
> 
> can you fix that for me?

I think easiest path is to force push the pre-mass-commit push as this branch

ojhunt commented:
I'm sure there's something you can do like `git push --force cb7e5ab58315e15eec03818161de1d71a41730da ...` 

But I'm a garbage git user so I would do:

```sh
git checkout cb7e5ab58315e15eec03818161de1d71a41730da
git branch -D mdenson:issue-32680
git checkout -b mdenson:issue-32680
git push --force --set-upstream <your remote> mdenson:issue-32680
```

mdenson commented:
i'm not git expert, but i think that is right

cor3ntin reviewed (approved):
LGTM, Thanks!
